src/braket/circuits/circuit.py

```diff
@@ -440,22 +440,20 @@ def _check_if_qubit_measured(
         Raises:
             ValueError: If adding a gate or noise operation after a measure instruction.
         """
-        if (
-            # check if there is a measure instruction on the target qubit
-            target
-            and target in self._measure_targets
-            # check if there is a measure instruction on any qubits in the target_mapping
-            or (target_mapping and any(targ in self._measure_targets for targ in target_mapping))
-            # If no target or target_mapping is supplied, check if there is a measure
-            # instruction on the current instructions target qubit
-            or (
-                instruction.target
-                and any(targ in self._measure_targets for targ in instruction.target)
+        if self._measure_targets:
+            measure_on_target = any(tar in self._measure_targets for tar in QubitSet(target))
+            measure_on_target_mapping = target_mapping and any(
+                targ in self._measure_targets for targ in target_mapping.values()
             )
-        ):
-            raise ValueError(
-                "cannot add a gate or noise operation on a qubit after a measure instruction."
+            measure_on_instruction_target = any(
+                targ in self._measure_targets for targ in QubitSet(instruction.target)
             )
+            if (
+                # check if there is a measure instruction on the targeted qubit(s)
+                measure_on_target_mapping
+                or any(tar in self._measure_targets for tar in QubitSet(target) + QubitSet(instruction.target))
```

Comment:
This doesn't actually work; we'd need to support addition on `QubitSet`s
```suggestion
                or any(tar in self._measure_targets for tar in QubitSet(target))
                or any(tar in self._measure_targets for tar in QubitSet(instruction.target))
```